frontend/components/admin/pages/AdvertisementEditor.tsx

```diff
@@ -0,0 +1,349 @@
+          title: '入力に誤りがあります',
+          status: 'error',
+          position: 'top',
+          duration: 3000,
+        })
+      } else {
+        toast({
+          title: responseData.message,
+          status: 'success',
+          position: 'top',
+          duration: 3000,
+        })
+      }
+    } catch (e) {
+      toast({
+        title: '広告情報の更新に失敗しました',
+        status: 'error',
+        position: 'top',
+        duration: 3000,
+      })
+    } finally {
+      SetIsSubmitting(false)
+    }
+  }
+  return (
+    <Box w="full" maxW="600px" mx="auto" p={6}>
+      <Stack spacing={4}>
+        <Link href="/admin/advertisement">
+          <Button colorScheme="green">一覧へ戻る</Button>
+        </Link>
+        <Box
+          p={4}
+          border="1px"
+          borderColor="gray.400"
+          borderRadius={9}
+          bg={THEME_COLOR.SECONDARY_WHITE}
+        >
+          <Text>広告ID：{advertisementData.id}</Text>
+        </Box>
+        <FormControl
+          isInvalid={
+            !!errors &&
+            errors.find &&
+            !!errors.find((e) => {
+              return e.path[0] === 'name'
+            })
+          }
+        >
+          <FormLabel>広告のタイトル</FormLabel>
+          <Input
+            type="text"
+            value={advertisementData.name}
+            onChange={(e) =>
+              setAdvertisementData({
+                ...advertisementData,
+                name: e.target.value,
+              })
+            }
+            aria-required={true}
+            bg={THEME_COLOR.SECONDARY_WHITE}
+            border="1px"
+            borderColor="gray.400"
+          />
+          <FormErrorMessage>
+            {errors &&
+              errors.find &&
+              errors.find((e) => {
+                return e.path[0] === 'name'
+              })?.message}
+          </FormErrorMessage>
+        </FormControl>
+        {/* <FormControl
+          isInvalid={
+            !!errors && errors.find && !!errors.find((e) => {
+              return e.path[0] === 'imageUrl'
+            })
+          }
+        >
+          <FormLabel>広告画像</FormLabel>
+          <Input
+            type="text"
+            value={advertisementData.imageUrl}
+            onChange={(e) =>
+              setAdvertisementData({
+                ...advertisementData,
+                imageUrl: e.target.value,
+              })
+            }
+            aria-required={true}
+            border="1px"
+            borderColor="gray.400"
+          />
+          <FormErrorMessage>
+            {
+              errors && errors.find && errors.find((e) => {
+                return e.path[0] === 'imageUrl'
+              })?.message
+            }
+          </FormErrorMessage>
+          TODO 画像が保存できるようになったら修正
+        </FormControl> */}
+        <FormControl
+          isInvalid={
+            !!errors &&
+            errors.find &&
+            !!errors.find((e) => {
+              return e.path[0] === 'author'
+            })
```

Comment:
```suggestion
            !!errors?.find((e) => {
              return e.path[0] === 'author'
            })
```

これでこれでエラー判定できませんか？
判定文が多いですね！

Reply:
他のバリデーション箇所も同様ですね！